src/audio/SDL_audiocvt.c

```diff
@@ -32,88 +32,85 @@
 
 #include "SDL_audio_resampler_filter.h"
 
-static int GetResamplerPaddingFrames(const int iinrate, const int ioutrate)
+static Sint64 GetResampleRate(const int src_rate, const int dst_rate)
 {
-    /* This function uses integer arithmetics to avoid precision loss caused
-     * by large floating point numbers. Sint32 is needed for the large number
-     * multiplication. The integers are assumed to be non-negative so that
-     * division rounds by truncation. */
-    const Sint32 inrate = (Sint32) iinrate;
-    const Sint32 outrate = (Sint32) ioutrate;
-    SDL_assert(inrate >= 0);
-    SDL_assert(outrate >= 0);
-    if (inrate == outrate) {
+    SDL_assert(src_rate > 0);
+    SDL_assert(dst_rate > 0);
+
+    if (src_rate == dst_rate)
         return 0;
-    } else if (inrate > outrate) {
-        return (int) (((RESAMPLER_SAMPLES_PER_ZERO_CROSSING * inrate) + (outrate - 1)) / outrate);
-    }
-    return RESAMPLER_SAMPLES_PER_ZERO_CROSSING;
+
+    return ((Sint64)src_rate << 32) / (Sint64)dst_rate;
+}
+
+static size_t GetResamplerAvailableOutputFrames(const size_t input_frames, const Sint64 resample_rate, const Sint64 resample_offset)
+{
+    SDL_assert(resample_offset < resample_rate);
```

Comment:
I am unlucky enough to trigger this assertion with `testaudiostreamdynamicresample`:
drag the slider back and forth between 4.5 and 5.0 (by shaking the mouse)
I have:
```
(gdb) print resample_offset
$1 = 9105173952
(gdb) print resample_rate
$2 = 9075802767
```.

Reply:
Should be fixed in the latest commit.

Reply:
I can't get it to crash anymore!
Great!